packages/core/components/ServerRender/__tests__/index.spec.tsx

```
@@ -102,4 +102,94 @@ describe("ServerRender", () => {
+
+  it("strips `children` and `dangerouslySetInnerHTML` before forwarding to `as`", () => {
+    const config: Config = {
+      components: {
+        Section: {
+          fields: {
+            content: { type: "slot" },
+          },
+          render: ({ content: Content }) => (
+            <Content
+              as="section"
+              dangerouslySetInnerHTML={{ __html: "<em>injected</em>" }}
```

Biome (2.5.6)

[error] Avoid passing both children and the dangerouslySetInnerHTML prop.

(lint/security/noDangerouslySetInnerHtmlWithChildren)